staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -405,17 +405,6 @@ func (s *GenericAPIServer) PrepareRun() preparedGenericAPIServer {
 	}
 	s.installReadyz()
 
-	// Register audit backend preShutdownHook.
-	if s.AuditBackend != nil {
-		err := s.AddPreShutdownHook("audit-backend", func() error {
-			s.AuditBackend.Shutdown()
```

Comment:
`AuditBackend.Shutdown` can no longer be executed as part of pre-shutdown hooks because of a deadlock scenario:
- `Shutdown` will block until `AuditBackend.Run` completes
- `AuditBackend.Run` stops when requests are drained 
- requests are drained when pre-shutdown hooks finish 

I added a fake audit backend to the test and i was able to reproduce the deadlock.